helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -389,29 +389,29 @@ receivers:
     priority: {{ $unit.priority }}
     operators:
     - type: add
-      field: $$resource["com.splunk.source"]
+      field: resource["com.splunk.source"]
       value: {{ $.Values.logsCollection.journald.directory }}
     - type: add
-      field: $$resource["com.splunk.sourcetype"]
+      field: resource["com.splunk.sourcetype"]
       value: 'EXPR("kube:journald:"+$$._SYSTEMD_UNIT)'
```

Comment:
`$$._SYSTEMD_UNIT` is equivalent to `$$body._SYSTEMD_UNIT`.
So, you must change it to `body._SYSTEMD_UNIT`